be/src/common/configbase.cpp

```diff
@@ -338,7 +340,11 @@ bool init(const char* conf_file, bool fill_conf_map, bool must_exist, bool set_t
         SET_FIELD(it.second, int32_t, fill_conf_map, set_to_default);
         SET_FIELD(it.second, int64_t, fill_conf_map, set_to_default);
         SET_FIELD(it.second, double, fill_conf_map, set_to_default);
-        SET_FIELD(it.second, std::string, fill_conf_map, set_to_default);
+        {
+            // add lock to ensure thread safe
+            std::lock_guard<std::mutex> lock(mutable_string_config_lock);
+            SET_FIELD(it.second, std::string, fill_conf_map, set_to_default);
+        }
```

Comment:
`init` is called in main when no other threads created, is it needed to use lock?

Reply:
> `init` is called in main when no other threads created, is it needed to use lock?

Thanks, you are right.